File: chanlib/src/main/java/com/mimireader/chanlib/models/ErrorChanThread.java
package com.mimireader.chanlib.models;

public class ErrorChanThread extends ChanThread {
    private final Throwable throwable;

    public ErrorChanThread(ChanThread chanThread, Throwable throwable) {
        this.boardName = chanThread.boardName;
        this.threadId = chanThread.threadId;
        this.posts.addAll(chanThread.posts);
        this.throwable = throwable;
    }

    public Throwable getError() {
        return throwable;
    }
}

File: chanlib/src/main/java/com/mimireader/chanlib/models/ErrorChanThread.kt
package com.mimireader.chanlib.models

import java.io.PrintWriter
import java.io.StringWriter

class ErrorChanThread(chanThread: ChanThread, val error: Throwable) : ChanThread() {

    init {
        boardName = chanThread.boardName
        threadId = chanThread.threadId
        posts.addAll(chanThread.posts)
    }

    override fun toString(): String {
        val sw = StringWriter()
        val pw = PrintWriter(sw)
        error.printStackTrace(pw)
        val trace: String = sw.toString()

        return "ErrorChanThread{board='$boardName',\ntitle='$boardTitle',\nthread id='$threadId',\npost count='${posts.size}',\nexception=${trace}}"
    }
}